### ISSUE TITLE
BUG: System redirects nonsensical

### ISSUE THREAD

**Issue by anisfeld** [prijatelilab/PrijateliTree#135]
## What Issue Are You Seeing or What Feature Needs To Be Added?

This issue pertains bizarre behavior in the system where the /round route is not going to the correct round. For example, in the picture below we see that game 32 has answers from round 1 for all 6 players and round two for others.  Yet, the player is directed to the screen as if they're in round 1!  This particular player already answered 2 rounds.

![Screenshot 2024-01-06 at 7 46 24 PM](https://github.com/prijatelilab/PrijateliTree/assets/25271399/0cb8ad2c-850c-4b3a-a290-c0abd797ad38)

The weird thing is that sometimes this player is directed to round 1 and sometimes round 2.  After going to the waiting screen ...

I have not been able to reproduce the error locally.

**Comment by anisfeld** [comment]
https://github.com/prijatelilab/PrijateliTree/assets/25271399/7f3e000f-9409-4f26-bc65-708bee795736

This is worse than I thought ... 
I think I'll first try to slow/reduce the number of hits to the database. Maybe using a singleton database is a bad idea though.

**Comment by anisfeld** [comment]
Print statements have me believe that  player.answers is sometimes failing to retrieve the data. I think this is the most reasonable culprit than alternative explanations described below. 

*this round should never be 0*
This implies that we see never `player.id == player_id` OR player.answers = []. From the database we know that player.answers had an observation (and that would be necessary to be sent to the waiting screen). 

```
2024-01-19T17:48:23.649106+00:00 app[web.1]: in all_set
2024-01-19T17:48:23.649192+00:00 app[web.1]: n_answers is 4 and len(game.players) is 6
2024-01-19T17:48:23.649192+00:00 app[web.1]: this round is 0
2024-01-19T17:48:23.649384+00:00 app[web.1]: INFO:     172.58.167.70:0 - "GET /games/52/player/310/all_set HTTP/1.1" 200 OK
2024-01-19T17:48:23.760144+00:00 app[web.1]: INFO:     172.58.167.70:0 - "GET /games/52/player/310/round HTTP/1.1" 200 OK
```



*the number of answers is shifting within the same player*

There should be 5 answers, but we see the response shift back and forth between 4 and 5. This also could happen if player.answers sometimes returns an empty set when it should not. 

Alternatively, max([n.round for n in player.answers]) might return 0 or worse.

```
2024-01-19T17:50:18.206187+00:00 app[web.1]: INFO:     172.58.167.70:0 - "GET /games/52/player/309/all_set HTTP/1.1" 200 OK
2024-01-19T17:50:20.206159+00:00 heroku[router]: at=info method=GET path="/games/52/player/309/all_set" host=intense-tundra-19419-c4bc16058534.herokuapp.com request_id=1aa2b688-4181-4ad7-ae9b-05634d10f9dd fwd="172.58.167.70" dyno=web.1 connect=0ms service=2ms status=200 bytes=177 protocol=http
2024-01-19T17:50:20.205428+00:00 app[web.1]: in all_set
2024-01-19T17:50:20.205516+00:00 app[web.1]: n_answers is 5 and len(game.players) is 6
2024-01-19T17:50:20.205519+00:00 app[web.1]: this round is 1

2024-01-19T17:50:20.205653+00:00 app[web.1]: INFO:     172.58.167.70:0 - "GET /games/52/player/309/all_set HTTP/1.1" 200 OK
2024-01-19T17:50:22.243574+00:00 heroku[router]: at=info method=GET path="/games/52/player/309/all_set" host=intense-tundra-19419-c4bc16058534.herokuapp.com request_id=00555389-733e-470e-bf78-39886c3868a2 fwd="172.58.167.70" dyno=web.1 connect=1ms service=2ms status=200 bytes=177 protocol=http

2024-01-19T17:50:22.242819+00:00 app[web.1]: in all_set
2024-01-19T17:50:22.242909+00:00 app[web.1]: n_answers is 5 and len(game.players) is 6
2024-01-19T17:50:22.242909+00:00 app[web.1]: this round is 1
2024-01-19T17:50:22.243046+00:00 app[web.1]: INFO:     172.58.167.70:0 - "GET /games/52/player/309/all_set HTTP/1.1" 200 OK

2024-01-19T17:50:24.289897+00:00 heroku[router]: at=info method=GET path="/games/52/player/309/all_set" host=intense-tundra-19419-c4bc16058534.herokuapp.com request_id=457ed998-24cf-4739-8370-7cf71301957d fwd="172.58.167.70" dyno=web.1 connect=8ms service=12ms status=200 bytes=177 protocol=http

2024-01-19T17:50:24.277671+00:00 app[web.1]: in all_set
2024-01-19T17:50:24.277681+00:00 app[web.1]: n_answers is 4 and len(game.players) is 6
2024-01-19T17:50:24.277681+00:00 app[web.1]: this round is 1
2024-01-19T17:50:24.277970+00:00 app[web.1]: INFO:     172.58.167.70:0 - "GET /games/52/player/309/all_set HTTP/1.1" 200 OK

2024-01-19T17:50:26.295302+00:00 app[web.1]: in all_set
2024-01-19T17:50:26.295390+00:00 app[web.1]: n_answers is 5 and len(game.players) is 6
2024-01-19T17:50:26.295391+00:00 app[web.1]: this round is 1
2024-01-19T17:50:26.295526+00:00 app[web.1]: INFO:     172.58.167.70:0 - "GET /games/52/player/309/all_set HTTP/1.1" 200 OK
```

**Comment by anisfeld** [comment]
Resolved with #145
